src/components/Home.jsx

```diff
@@ -10,17 +12,30 @@ import ListItemText from '@mui/material/ListItemText';
 import ListItemAvatar from '@mui/material/ListItemAvatar';
 import Avatar from '@mui/material/Avatar';
 import Typography from '@mui/material/Typography';
-import Divider from '@mui/material/Divider';
 import {drawerWidth} from '../components/Header'
  
 
 export default function Home() {
+  const location = useLocation()
+  const searchParams = new URLSearchParams(location.search)
+  const topic = searchParams.get('topic')
+  const [sortBy, setSortBy] = useState('created_at');
+  const [order, setOrder] = useState('DESC');
+  const [newSort, setNewSort] = useState(false)
   const [articles, setArticles] = useState([])
+
   useEffect(() => {
-      getArticles().then((response) => {
+    if (topic) {
+      getTopicalArticles(topic, sortBy, order).then((response) => {
+        setArticles(response.data.articles)
+    })
+    } else {
+      getArticles(sortBy, order).then((response) => {
           setArticles(response.data.articles)
       })
-  }, [])
+    }
+    setNewSort(false)
+  }, [topic, newSort])
```

Comment:
newSort as dependency is not working as expected. But it works when putting sortBy and order in.